File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int lottos[], size_t lottos_len, int win_nums[], size_t win_nums_len) {
    int* answer = (int*)malloc(2*sizeof(int));
    int same = 0;
    int zeros = 0;
    for(int i=0;i<lottos_len;i++){
        if(lottos[i] == 0){zeros++;continue;}
        for(int j=0;j<win_nums_len;j++){
            if(lottos[i] == win_nums[j]){same++;}
        }
    }
    answer[1] = 7 - same;
    answer[0] = answer[1] - zeros;
    if(answer[1] == 7){answer[1] = 6;}
    if(answer[0] == 7){answer[0] = 6;}
    
    return answer;
}

File: 프로그래머스/1/77484.로또의 최고 순위와 최저 순위-Python/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    already = 0
    zeros = 0
    for lo in lottos:
        if lo in win_nums:
            already += 1
        if 0 == lo:
            zeros += 1

    b = 7 - already if already > 1 else 6
    a = 7 - already - zeros if already + zeros > 1 else 6
    
    return [a,b]